File: 1173-34-2364-count-number-of-bad-pairs/1173-34-2364-count-number-of-bad-pairs.py
from collections import defaultdict

class Solution:
    def countBadPairs(self, nums):
        n = len(nums)
        bads = n * (n - 1) // 2
        mp = defaultdict(int)
        for i in range(n):
            bads -= mp[nums[i] - i]
            mp[nums[i] - i] += 1

        return bads

File: 1173-34-2364-count-number-of-bad-pairs/1173-34-2364-count-number-of-bad-pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
        int n = nums.size();
        long long bads = n * (n-1LL)/2;
        unordered_map<int,int>mp;
        for(int i = 0 ; i < n;i++)
            bads-=mp[nums[i]-i]++;
        return bads;
    }
};